storage/db-tool/src/replay_on_archive.rs

```diff
@@ -204,6 +204,13 @@ impl Verifier {
         let mut expected_txn_infos = Vec::new();
         let mut chunk_start_version = start;
         for (idx, item) in txn_iter.enumerate() {
+            // timeout check
+            if let Some(duration) = self.timeout_secs {
+                if self.replay_stat.get_elapsed_secs() >= duration {
+                    return Ok(total_failed_txns);
```

Comment:
timeout is an okay, not an error?

Reply:
Timeout here is to force the worker to stop replaying and return the existing results immediately.  error would discard the all the results and can be a waste

Reply:
Why not use something like tokio::timeout here instead of making our own timeout?

If we dont enforce a timeout on a future using tokio (which also has its own caveats) then it seems like this might never actually time out here?